### DISCUSSION TITLE
Default for mapping

### DISCUSSION THREAD

**Discussion by chakrit** [cue-lang/cue#1126] [Q&A] [answered]
I am trying to model a list of personnel data and roles in cue.

* For _some_ job titles, a specific role is assigned.
* For other job titles, a default role is assigned.

Now I have a "people" data source coming from an API like this:

```cue
people: [
  { name: "Jack", job: "CEO", username: "jack" },
  { name: "John", job: "Admin", username: "john" },
  { name: "Jill", job: "HR", username: "jill" }
  // there are ~20 diff jobs
]
```

And I'd like to map them to a "role" in another system based on a mapping:

```cue
#Role: "admin" | "viewer" | "commenter" | "no_access"

role_map: [string]: #Role
role_map: {
  "CEO": "admin"
  "Admin": "commenter"
}

role_config: [
  for p in people if len(p.username) > 0 {
    username: p.username,
    role: role_map[p.job] | *"no_access"
  }
]
```

However this causes a `incomplete cause disjunction` error on the `role:` line. It seems I'm still mixing types and values incorrectly (yes, everything's a value but still) What am I missing here?

The behavior I want is simply that if a mapping from a person's job to a role is not specified, the output would still has the person's entry but with role set to the default role.

**Desired Output:**
```cue
[ { username: "jack", role: "admin" }
, { username: "john", role: "commenter" }
, { username: "jill", role: "no_access" }
]
```

**Comment by verdverm** [comment, marked as answer]
`role_map: {string: #Role}`, what are you trying to achieve here? TBH, I'm surprised this is allowed (i.e. having a keyword be a label without quotes). Have a look at this example: https://cuelang.org/play/?id=17Du0HMdzco#cue@export@cue

I think what you want is `role_map: [string]: #Role`

**Reply by chakrit** [reply]
I mistyped that while trying to simplify the code I have. I updated and the result are the same: `incomplete cause disjunction`

**Reply by jlongtine** [reply]
@chakrit I think you need to change the default to `role: *role_map[p.job] | "no_access"`

https://cuelang.org/play/?id=gz9ak6pXcuT#cue@export@cue

**Reply by chakrit** [reply]
@jlongtine that works 👍 . seems i still don't understand how `*` works in this case.

And somehow on the latest version and on play ground it no longer reports `incomplete cause disjunction` now.
